zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/DeploymentCreateProcessor.java

```diff
@@ -149,6 +149,9 @@ public ProcessingError tryHandleError(
 
   private void transformAndDistributeDeployment(final TypedRecord<DeploymentRecord> command) {
     final DeploymentRecord deploymentEvent = command.getValue();
+    final long key = keyGenerator.nextKey();
+    deploymentEvent.setKey(key);
+
```

Comment:
> My feeling is that it's useful to add the `deploymentKey` to the deployment record value, because it serves a purpose now. Perhaps in the future we'll also want to allow users to delete resources by deployment key, rather than by resource key specifically.

Yes, I would agree. We might also enhance the deployment concept further by enabling named deployments (as in C7) and/or showing deployments in Operate. So, if there's no clear reason not to do it, I would keep the `deploymentKey` in the record.